contextproject-ms3/src/main/java/contextproject/sorters/MaxFlow.java

```diff
@@ -0,0 +1,149 @@
+package contextproject.sorters;
+
+import contextproject.models.Playlist;
+import contextproject.models.Track;
+
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.alg.EdmondsKarpMaximumFlow;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MaxFlow {
+
+  private DirectedGraph<Track, WeightedEdge> graph;
+  private Map<WeightedEdge, Double> bestFlow;
+  private Track bestSource;
+  private Track bestSink;
+  private TrackTree trackTree;
+  private Playlist optimalPath;
+
+  /**
+   * Constructor for a MaxFlow calculator.
+   * @param graph Graph
+   */
+  public MaxFlow(Graph graph) {
+    this.graph = graph;
+    calculateMaxflow();
+    convertMap();
+    calculateOptimalPath();
+  }
+
+  /**
+   * calculate the maximum flow from graph.
+   */
+  private void calculateMaxflow() {
+    Double bestscore = 0.0;
+
+    for (int i = 0; i < graph.vertexSet().size(); i++) {
+      for (int j = 0; j < graph.vertexSet().size(); j++) {
+        if (i != j) {
+          Track source = (Track) graph.vertexSet().toArray()[i];
+          Track sink = (Track) graph.vertexSet().toArray()[j];
+          WeightedEdge edge1 = graph.getEdge(source, sink);
+          WeightedEdge edge2 = graph.getEdge(sink, source);
+          
+          graph.removeEdge(edge1);
+          graph.removeEdge(edge2);
+          
+          EdmondsKarpMaximumFlow<Track, WeightedEdge> edmondsKarp = new 
+              EdmondsKarpMaximumFlow<Track, WeightedEdge>(graph);
+          edmondsKarp.calculateMaximumFlow(source, sink);
+          
+          graph.addEdge(source, sink, edge1);
+          graph.addEdge(sink, source, edge2);
+          
+          if (edmondsKarp.getMaximumFlowValue() > bestscore) {
+            bestscore = edmondsKarp.getMaximumFlowValue();
+            bestFlow = edmondsKarp.getMaximumFlow();
+            bestSource = edmondsKarp.getCurrentSource();
+            bestSink = edmondsKarp.getCurrentSink();
+          }
+
+        }
+
+      }
+    }
+  }
```

Comment:
Can you add tests with some very basic/small examples? Now there's no way to guarantee this works.